### ISSUE TITLE
Add start as a modifier in InputFieldSelector

### ISSUE THREAD

**Issue by theGC** [processwire/processwire-issues#1259]
### Short description of the issue

`start` is not available as a modifier within InputFieldSelector

### Expected behavior

`start` to be available

### Actual behavior

it isn't

### Optional: Screenshots/Links that demonstrate the issue

![Screenshot 2020-10-13 at 03 17 28](https://user-images.githubusercontent.com/1188840/95807713-c0c28000-0d02-11eb-85ba-85216e98d08f.png)

### Optional: Suggestion for a possible fix

Add `start` modifier to line 588 of wire/modules/Inputfield/InputfieldSelector/InputfieldSelector.module

https://github.com/theGC/processwire/blob/48092d503e90720df796adb0a5f8bfd3afd84889/wire/modules/Inputfield/InputfieldSelector/InputfieldSelector.module#L588

```php
'start' => array(
  'input' => 'integer',
  'label' => $this->_('Start (requires limit)'),
  'operators' => array('=')
),
```

### Steps to reproduce the issue

install processwire, add a selector field to a page

### Setup/Environment

any

**Comment by theGC** [comment]
Also worth adjusting `renderTestSelector` to allow for the new start modifier:

https://github.com/theGC/processwire/blob/65092e06acd31738122aac95812dfde2b9c9a57f/wire/modules/Inputfield/InputfieldSelector/InputfieldSelector.module#L751

```php
// take into account a limit=n
if(strpos($selector, 'limit=') !== false && preg_match('/\blimit=(\d+)/', $selector, $matches)) {
    $out = ' ' . sprintf($this->_('(%d without limit)'), $cnt);
    // take into account a start=n
    if(strpos($selector, 'start=') !== false && preg_match('/\bstart=(\d+)/', $selector, $startMatches)) {
        $cnt -= $startMatches[1];
        if($cnt < 0) $cnt = 0;
    }
    if($cnt > $matches[1]) $cnt = $matches[1];
}
```

**Comment by ryancramerdesign** [comment]
@theGC This one was intentionally left out since it's the current pagination number (`$input->pageNum`) that determines and sets the `start=` value. The primary public API usage of it is so that you can specify `start=0` to override automatic pagination for things like top level navigation that doesn't change on paginations. I wasn't aware there was a use case for having 'start' in this module. Can you describe more about the use case?

**Comment by theGC** [comment]
yeah sure, we're using it to allow clients to select pages (categories or products), showing a limited set of these, starting at a certain index.

The above is set up within a matrix repeater, I've screengrabbed a sample set up below:

![Screenshot 2020-10-19 at 15 28 13](https://user-images.githubusercontent.com/1188840/96464589-ea5a3a80-121f-11eb-9862-f1fe0faabb65.png)

The above allows the user to create blocks of content, with the same selector broken up over 2 or more parts of the page.

**Comment by matjazpotocnik** [comment]
@ryancramerdesign 

Proposed fix adds `start` to the visual selector builder using the existing integer input and equals operator, with the label **Start (requires limit)**.

It also updates the preview count to account for the offset before applying the limit. For example, with 12 matching pages, `start=10, limit=5` displays 2 matches; an offset beyond the results displays 0.

The preview reads top-level selectors rather than matching modifier-like text inside quoted values. Explicit `start=0` remains supported and overrides automatic pagination. Without a positive limit, the preview shows the total; `limit=0` is treated as unlimited. Negative-limit and advanced `_custom` preview behavior are outside this patch's scope.

No JavaScript changes are required by the proposed implementation: the existing AJAX and reload paths use the same modifier definitions and renderer.

## Diff

```diff
diff --git a/wire/modules/Inputfield/InputfieldSelector/InputfieldSelector.module b/wire/modules/Inputfield/InputfieldSelector/InputfieldSelector.module
index 5ae2f77..b56f551 100644
--- a/wire/modules/Inputfield/InputfieldSelector/InputfieldSelector.module
+++ b/wire/modules/Inputfield/InputfieldSelector/InputfieldSelector.module
@@ -606,6 +606,11 @@ class InputfieldSelector extends Inputfield implements ConfigurableModule {
 				'label' => $this->_('Limit'),
 				'operators' => array('=')
 			),
+			'start' => array(
+				'input' => 'integer',
+				'label' => $this->_('Start (requires limit)'),
+				'operators' => array('=')
+			),
 			'sort' => array(
 				'input' => 'select',
 				'label' => $this->_('Sort'),
@@ -768,10 +773,18 @@ class InputfieldSelector extends Inputfield implements ConfigurableModule {
 			$selector = $this->sanitizeSelectorString($selector); 
 			$cnt = $this->wire()->pages->count($selector, array('allowCustom' => true));
 			$out = '';
-			// take into account a limit=n 
-			if(strpos($selector, 'limit=') !== false && preg_match('/\blimit=(\d+)/', $selector, $matches)) {
-				$out = ' ' . sprintf($this->_('(%d without limit)'), $cnt); 
-				if($cnt > $matches[1]) $cnt = $matches[1]; 
+			// Read top-level modifiers, not matching text within selector values.
+			$limit = 0;
+			$start = 0;
+			foreach(new Selectors($selector) as $s) {
+				if($s->field === 'limit') $limit = (int) $s->value;
+				if($s->field === 'start') $start = (int) $s->value;
+			}
+			if($limit > 0) {
+				$label = $start ? $this->_('(%d without start/limit)') : $this->_('(%d without limit)');
+				$out = ' ' . sprintf($label, $cnt);
+				if($start < 0) $start = max(0, $cnt + $start);
+				$cnt = min($limit, max(0, $cnt - $start));
 			}
 			$out = sprintf($this->_n('matches %d page', 'matches %d pages', $cnt), $cnt) . $out;
 			if(self::debug) $out .= " (" . $selector . ")";
```
